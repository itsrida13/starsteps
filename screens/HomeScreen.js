import React from 'react';
import { SafeAreaView, StyleSheet, Image } from 'react-native';
import { Button, Text } from 'react-native-paper';

export default function HomeScreen({ navigation }) {
  const handleStart = () => {
    console.log('Start button pressed!');
    // navigation.navigate('NextScreen'); // later
  };

  return (
    <SafeAreaView style={styles.container}>
      <Image
        source={require('../assets/starsteps-logo.png')}
        style={styles.logo}
      />
      <Text style={styles.title}>Welcome to StarSteps ðŸŒŸ</Text>
      <Text style={styles.subtitle}>
        Learning and fun for every amazing child!
      </Text>
      <Button
        mode="contained"
        onPress={handleStart}
        style={styles.startButton}
        labelStyle={styles.buttonText}
      >
        Start Learning
      </Button>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFE4F0',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 20,
  },
  logo: {
    width: 160,
    height: 160,
    marginBottom: 30,
    resizeMode: 'contain',
  },
  title: {
    fontSize: 28,
    fontWeight: 'bold',
    color: '#3B3B98',
    marginBottom: 10,
    textAlign: 'center',
  },
  subtitle: {
    fontSize: 18,
    color: '#6D6D6D',
    marginBottom: 40,
    textAlign: 'center',
  },
  startButton: {
    backgroundColor: '#FF7EB3',
    borderRadius: 30,
    paddingHorizontal: 30,
    paddingVertical: 10,
  },
  buttonText: {
    fontSize: 18,
    color: 'white',
  },
});
