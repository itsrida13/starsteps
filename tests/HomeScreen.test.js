import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import HomeScreen from '../screens/HomeScreen';

describe('HomeScreen', () => {
  it('renders welcome text and button', () => {
    const { getByText } = render(<HomeScreen />);

    expect(getByText('Welcome to StarSteps ðŸŒŸ')).toBeTruthy();
    expect(getByText('Start Learning')).toBeTruthy();
  });

  it('responds to button press', () => {
    const { getByText } = render(<HomeScreen />);
    const button = getByText('Start Learning');
    
    fireEvent.press(button);
    // Currently it only logs to console; nothing to assert
  });

  // âœ… Snapshot test
  it('matches the snapshot', () => {
    const { toJSON } = render(<HomeScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
