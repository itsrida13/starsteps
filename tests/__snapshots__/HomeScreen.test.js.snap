// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen matches the snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFE4F0",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/starsteps-logo.png",
      }
    }
    style={
      {
        "height": 160,
        "marginBottom": 30,
        "resizeMode": "contain",
        "width": 160,
      }
    }
  />
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "color": "#3B3B98",
          "fontSize": 28,
          "fontWeight": "bold",
          "marginBottom": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Welcome to StarSteps ðŸŒŸ
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "color": "#6D6D6D",
          "fontSize": 18,
          "marginBottom": 40,
          "textAlign": "center",
        },
      ]
    }
  >
    Learning and fun for every amazing child!
  </Text>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#FF7EB3",
        "borderRadius": 30,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#FF7EB3",
          "borderColor": "transparent",
          "borderRadius": 30,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flex": undefined,
          "minWidth": 64,
          "paddingHorizontal": 30,
          "paddingVertical": 10,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 30,
            },
          ]
        }
        testID="button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  [
                    {
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    false,
                    {
                      "marginHorizontal": 24,
                      "marginVertical": 10,
                    },
                    undefined,
                    false,
                    {
                      "color": "rgba(255, 255, 255, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    {
                      "color": "white",
                      "fontSize": 18,
                    },
                  ],
                ],
              ]
            }
            testID="button-text"
          >
            Start Learning
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
